### PR TITLE
Ptkfix

### LINKED ISSUE #753: Out-of-date prompt-toolkit causes `TypeError` crash on start
Since upgrading to 0.2.7, running `xonsh` (with or without `--no-rc`) gives me a `TypeError`:

```
$ xonsh --version
('xonsh/0.2.7',)
$ xonsh --no-rc
/usr/local/lib/python3.4/dist-packages/xonsh/shell.py:97: RuntimeWarning: prompt-toolkit version < v0.57 and may not work as expected. Please update.
  warn(msg, RuntimeWarning)
Traceback (most recent call last):
  File "/usr/local/bin/xonsh", line 9, in <module>
    load_entry_point('xonsh==0.2.7', 'console_scripts', 'xonsh')()
  File "/usr/local/lib/python3.4/dist-packages/xonsh/main.py", line 214, in main
    shell.cmdloop()
  File "/usr/local/lib/python3.4/dist-packages/xonsh/ptk/shell.py", line 112, in cmdloop
    line = self.singleline(auto_suggest=auto_suggest)
  File "/usr/local/lib/python3.4/dist-packages/xonsh/ptk/shell.py", line 81, in singleline
    display_completions_in_columns=multicolumn)
  File "/usr/local/lib/python3.4/dist-packages/xonsh/ptk/shortcuts.py", line 71, in prompt
    application=create_prompt_application(message, **kwargs),
TypeError: create_prompt_application() got an unexpected keyword argument 'reserve_space_for_menu'
```

`xonsh --shell-type readline` starts successfully, so I'm guessing the crash is caused by the prompt-toolkit version. I can't remember how I installed prompt-toolkit, so I'll stick with readline for a bit.

I'm running Ubuntu 14.04 x86_64, kernel 3.16.0-67-generic. `xonsh` is installed system-wide using `pip3`. Feel free to ask for any more information you need.

### PR DESCRIPTION
**Pull request by scopatz**
Should fix #753

### CODE REVIEW

File: xonsh/ptk/shortcuts.py [hunk 1]
```diff
@@ -64,6 +64,8 @@ def prompt(self, message='', **kwargs):
 
         # Create CommandLineInterface.
         if self.cli is None:
+            if self.major_minor < (0, 57):
+                kwargs.pop('reserve_space_for_menu', None)
```
Comment (melund) [inline xonsh/ptk/shortcuts.py]: The PTK change log says it was introduced in 0.55. But I guess it shouldn't matter. Else this PR seems fine to me. @scopatz Should wait for response on the issue before we merge?

Comment (scopatz) [inline xonsh/ptk/shortcuts.py]: I think that this should probably still be merged, even if there is no response to the other issue.  This is still fixes a bug.

Comment (scopatz) [inline xonsh/ptk/shortcuts.py]: Oh, and ptk didn't report its version number until v0.57 :)

**Comment by melund** [comment]
Ok. This is from before appveyor stop working...I will just go ahead and merge this in.

**Pull request merged**